core/vm/evm.go

```diff
@@ -553,7 +629,7 @@ func (evm *EVM) Create(caller ContractRef, code []byte, gas uint64, value *big.I
 //
 // The different between Create2 with Create is Create2 uses sha3(0xff ++ msg.sender ++ salt ++ sha3(init_code))[12:]
 // instead of the usual sender-and-nonce-hash as the address where the contract is initialized at.
-func (evm *EVM) Create2(caller ContractRef, code []byte, gas uint64, endowment *big.Int, salt *uint256.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
+func (evm *EVM) Create2(caller *Contract, code []byte, gas uint64, endowment *big.Int, salt *uint256.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
```

Comment:
what is the intent behind this change?

Reply:
Should have been removed.  That was a hold-over from when I had gas-charging in the EVM opcode implementations (because `ContractRef` doesn't give you a method to charge gas, but the actual value type passed to the method is always `Contract*`